lib/taurus/core/taurusfactory.py

```diff
@@ -217,7 +217,8 @@ def getAttribute(self, name):
         try:
             # this works only if the devname is present in the attr full name
             # (not all schemes are constructed in this way)
-            devname = v.getUriGroups(fullname)['devname']
+            groups = v.getUriGroups(fullname)
+            devname = '{0}:{1}'.format(groups['scheme'], groups['devname'])
```

Comment:
I am not sure this is the right way to deal with this: in principle, XXXFactory.getDevice (and other methods for getting Attribute, Database, etc.) should assume that the scheme is the one of the XXXFactory itself (so it should not require thatthe scheme is passed explicitly)

Reply:
Agree. It seems it is not need